contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,335 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 #include "contrib/endpoints/src/api_manager/check_security_rules.h"
+#include <iostream>
+#include <sstream>
+#include "contrib/endpoints/src/api_manager/auth/lib/json_util.h"
+#include "contrib/endpoints/src/api_manager/proto/security_rules.pb.h"
+#include "contrib/endpoints/src/api_manager/utils/marshalling.h"
 
-#include <string>
-
-#include "contrib/endpoints/include/api_manager/api_manager.h"
-#include "contrib/endpoints/include/api_manager/request.h"
-
+using ::google::api_manager::auth::GetProperty;
+using ::google::api_manager::auth::GetStringValue;
 using ::google::api_manager::utils::Status;
+using ::google::protobuf::Map;
+using ::google::protobuf::util::error::Code;
 
 namespace google {
 namespace api_manager {
-
 namespace {
 
 const char kFirebaseServerStaging[] =
     "https://staging-firebaserules.sandbox.googleapis.com/";
 
+const char kFirebaseService[] =
+    "/google.firebase.rules.v1.FirebaseRulesService";
+
+const char kFailedFirebaseReleaseFetch[] = "Failed to fetch Firebase Release";
+const char kFailedFirebaseTest[] = "Failed to execute Firebase Test";
+const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
+const char kTestSuccess[] = "SUCCESS";
+
+const std::string GetFirebaseServer(context::RequestContext &context) {
+  return context.service_context()
+      ->config()
+      ->server_config()
+      ->api_check_security_rules_config()
+      .firebase_server();
+}
+
+void SetProtoValue(const std::string &key,
+                   const ::google::protobuf::Value &value,
+                   ::google::protobuf::Value *head) {
+  ::google::protobuf::Struct *s = head->mutable_struct_value();
+  Map<std::string, google::protobuf::Value> *fields = s->mutable_fields();
+  (*fields)[key] = value;
+}
+
+const std::string GetReleaseName(context::RequestContext &context) {
+  return context.service_context()->service_name() + ":" +
+         context.service_context()->service().apis(0).version();
+}
+
+const std::string GetRulesetTestUri(context::RequestContext &context,
+                                    const std::string &ruleset_id) {
+  return GetFirebaseServer(context) + "v1/" + ruleset_id + ":test?alt=json";
+}
+
+const std::string GetReleaseUrl(context::RequestContext &context) {
+  return GetFirebaseServer(context) + "v1/projects/" +
+         context.service_context()->project_id() + "/releases/" +
+         GetReleaseName(context);
+}
+
+std::string GetOperation(std::string &httpMethod) {
+  if (httpMethod == "POST") {
+    return std::string("create");
```

Comment:
Done. Some parts of the code were using them as I have done. I did not know this was a strict rule.